src/cmdexpand.c

```diff
@@ -2127,6 +2128,34 @@ set_context_in_scriptnames_cmd(expand_T *xp, char_u *arg)
 }
 #endif
 
+/*
+ * Set the completion context for the :filetype command. Always returns NULL.
+ */
+    static char_u *
+set_context_in_filetype_cmd(expand_T *xp, char_u *arg)
+{
+    char_u *p;
+
+    xp->xp_context = EXPAND_FILETYPECMD;
+    xp->xp_pattern = arg;
+
+    p = skipwhite(arg);
+    if (*p == NUL)
+	return NULL;
+
+    if (STRNCMP("indent ", p, 7) == 0 || STRNCMP("plugin ", p, 7) == 0)
```

Comment:
Although strange, it's possible that a TAB follows `indent` or `plugin`. Use `skiptowhite()`?

Reply:
Also, `filetype` can accept both `plugin` and `indent`:
```c
    // Accept "plugin" and "indent" in any order.
    for (;;)
    {
	if (STRNCMP(arg, "plugin", 6) == 0)
	{
	    plugin = TRUE;
	    arg = skipwhite(arg + 6);
	    continue;
	}
	if (STRNCMP(arg, "indent", 6) == 0)
	{
	    indent = TRUE;
	    arg = skipwhite(arg + 6);
	    continue;
	}
	break;
    }
```
E.g. `:filetype plugin indent on`, so `indent` should still be completed after `plugin`.

Reply:
ah, right, but that makes returning the list of completions a bit more complex.